code/modules/almayer/weapons.dm

```diff
@@ -0,0 +1,135 @@
+/**
+ * Space weapons it's self for ship to ship or PKO/Xeno PKO things
+ */
+/datum/space_weapon
+	var/name = "SMP"
+	var/list/possibly_ammunition = list()
+	//add some useful things here and make it object... later... skill issue.
+
+/datum/space_weapon/proc/on_shot(location, list/potential_ammo, intercept_chance)
+	var/intercepted = 0
+	var/missed = 0
+	var/hits = 0
+	for(var/turf/picked_atom in location)
+		var/datum/space_weapon_ammo/ammo = GLOB.space_weapons_ammo[pick(potential_ammo)]
+		var/accuracy = rand(1, 100)
+		if(ammo.interceptable && intercept_chance > accuracy)
+			ammo.miss_target(picked_atom, TRUE)
+			intercepted++
+		else if(ammo.base_miss_chance + ammo.interceptable ? intercept_chance : 0 > accuracy)
+			ammo.miss_target(picked_atom, FALSE)
+			missed++
+		else
+			ammo.hit_target(picked_atom)
+			hits++
+	shipwide_ai_announcement("WARNING, [hits] HIT SHIP HULL, [missed] MISSED AND [intercepted] INTERCEPTED!", MAIN_AI_SYSTEM, 'sound/effects/double_klaxon.ogg')
+
+/datum/space_weapon/proc/shot_message(quantity, hit_eta)
+	return
+
+/datum/space_weapon/rail_gun
+	name = "Railgun"
+	possibly_ammunition = list("Piercing Near-Lightning Railgun Projectile", "Piercing Near-Lightning Railgun Projectile of Increased Strength")
```

Comment:
It would be better to just grab the subtypes of /datum/space_weapon_ammo/rail_gun rather than hard coding as a list of strings. The datum would be more useful to get information. Also the variable name is a little odd.

E.g. `possible_ammunition = subtypesof(/datum/space_weapon/rail_gun)` but note that this will instead be storing datum paths

subtypes of is just https://www.byond.com/docs/ref/#/proc/typesof minus the base type.